app/components/RegisterSingleTree/MapMarking.js

```diff
@@ -254,62 +254,162 @@ class MapMarking extends React.Component {
-              disabled={loader}
-              btnText={i18next.t('label.tree_map_marking_btn')}
-              style={styles.bottomBtnWith}
-            />
+      <Modal transparent visible={isAccuracyModalShow}>
+        <View style={styles.modalContainer}>
+          <View style={styles.contentContainer}>
+            <Text
+              style={{
+                color: '#000000',
+                fontFamily: Typography.FONT_FAMILY_BOLD,
+                fontSize: Typography.FONT_SIZE_18,
+                paddingBottom: 18,
+              }}>
+              GPS Accuracy
+            </Text>
+            <Text style={styles.accuracyModalText}>
+              You can use the GPS data to determine accuracy of a location. Tree Mapper only allows
+              on site registration if location is accurate to 30 meter
+            </Text>
+            <Text style={styles.accuracyModalText}>
+              <Text style={{ color: '#87B738', fontFamily: Typography.FONT_FAMILY_BOLD }}>
+                Green
+              </Text>{' '}
+              = Accurate to 10 meter
+            </Text>
+            <Text style={styles.accuracyModalText}>
+              <Text style={{ color: '#CBBB03', fontFamily: Typography.FONT_FAMILY_BOLD }}>
+                Yellow
+              </Text>{' '}
+              = Accurate to 30 meter
+            </Text>
+            <Text style={styles.accuracyModalText}>
+              <Text style={{ color: '#FF0000', fontFamily: Typography.FONT_FAMILY_BOLD }}>Red</Text>{' '}
+              = Greater than 30 meter
+            </Text>
+            <TouchableOpacity
+              style={{
+                alignSelf: 'center',
+                paddingTop: 25,
+              }}>
+              <Text
+                style={{
+                  color: '#87B738',
+                  fontFamily: Typography.FONT_FAMILY_REGULAR,
+                  fontSize: Typography.FONT_SIZE_14,
+                }}
+                onPress={() => setIsAccuracyModalShow(false)}>
+                Close
+              </Text>
+            </TouchableOpacity>
           </View>
         </View>
-        <LinearGradient style={styles.headerCont} colors={[Colors.WHITE, 'rgba(255, 255, 255, 0)']}>
-          <SafeAreaView />
-          <Header
-            onBackPress={this.onPressBack}
-            closeIcon
-            headingText={i18next.t('label.tree_map_marking_header')}
+      </Modal>
+    );
+  };
+
+  const renderConfirmationModal = () => {
+    return (
+      <AlertModal
+        visible={isAlertShow}
+        heading={'Poor Accuracy'}
+        message={
+          'Your location accuracy is POOR, which is not recommended. Are you sure you want to continue?'
+        }
+        primaryBtnText={'Try Again'}
+        secondaryBtnText={'Continue'}
+        onPressPrimaryBtn={onPressTryAgain}
+        onPressSecondaryBtn={onPressAlertContinue}
+      />
+    );
+  };
+
+  const renderLocationAlert = () => {
+    return (
+      <AlertModal
```

Comment:
When installing the iOS app newly, I get this alert and are not able to go to my location specified for my iPhone 11 simulator:

![Simulator Screen Shot - iPhone 11 - 2021-02-23 at 13 36 40](https://user-images.githubusercontent.com/1532418/108847612-32c7a180-75e0-11eb-80f4-e7052eff61c4.png)

Reply:
I see `{"code": 1, "message": "Location permission denied"} updateCurrentPosition` in my logs.

Reply:
Despite the iOS settings show me that location access is granted, in the app I still get this dialog with the error `{"code": 2, "message": "Unable to retrieve location"} updateCurrentPosition` in the logs.

Reply:
Sorry, it was because my iOS simulator was configured the way it really had no location, so showing the error was correct.